CefSharp.Offscreen.Direct3D.Example/D3D11Renderer.cs

```diff
@@ -0,0 +1,408 @@
+using SampleDescription = SharpDX.DXGI.SampleDescription;
+using Vector2 = SharpDX.Vector2;
+using Vector4 = SharpDX.Vector4;
+using Viewport = SharpDX.Viewport;
+
+namespace CefSharp.Offscreen.Direct3D.Example;
+
+/// <summary>
+/// D3D11 renderer for CefSharp.Offscreen.Direct3D.Example
+/// </summary>
+public class D3D11Renderer
+{
+    private Texture2D _sharedTexture;
+    private Device _device;
+    private Adapter _adapter;
+    private static VertexDX11[] s_vertices;
+    private int _width;
+    private int _height;
+    private VertexBufferBinding _binding;
+    private Texture2D _popupLayer;
+    private Point _popupPosition;
+    private const string ShaderSrc = "Texture2D tex;   \n" +
+                                     "   \n" +
+                                     "SamplerState texSampler   \n" +
+                                     "{   \n" +
+                                     "   Texture = <tex>;   \n" +
+                                     "   Filter = MIN_MAG_MIP_POINT; \n" +
+                                     "};   \n" +
+                                     "   \n" +
+                                     "struct PS_IN   \n" +
+                                     "{  \n" +
+                                     "   float4 pos : SV_POSITION; \n" +
+                                     "   float2 Tex : TEXCOORD;  \n" +
+                                     "};   \n" +
+                                     "   \n" +
+                                     "PS_IN vertex( PS_IN input )    \n" +
+                                     "{  \n" +
+                                     "   return input;   \n" +
+                                     "}  \n" +
+                                     "   \n" +
+                                     "float4 pixel( PS_IN input ) : SV_Target  \n" +
+                                     "{  \n" +
+                                     "   return tex.Sample( texSampler, input.Tex );   \n" +
+                                     "}  \n" +
+                                     "   \n";
+
+    [StructLayout(LayoutKind.Sequential)]
+    private struct VertexDX11
+    {
+        public Vector4 Position;
+        public Vector2 TexCoord0;
+
+        public static InputElement[] GetInputElements()
+        {
+            InputElement[] inputElements = [new("SV_POSITION", 0, Format.R32G32B32A32_Float, 0, 0), new("TEXCOORD", 0, Format.R32G32_Float, 16, 0)];
+            return inputElements;
+        }
+    }
+
+    private void CreateDevice()
+    {
+        CreateAdapter();
+        UpgradeAdapter();
+        _device = new Device(_adapter, DeviceCreationFlags.BgraSupport, FeatureLevel.Level_11_1);
+        
+        UpgradeDevice();
+    }
+
+    private void CreateAdapter()
+    {
+        using var f = new Factory1();
+        _adapter = f.GetAdapter(0);
+    }
+
+    public D3D11Renderer(int windowWidth, int windowHeight)
+    {
+        Cef.ShutdownStarted += (_, _) => Destroy();
```

Comment:
Given `D3D11Renderer` appears create an instance per `ChromiumWebBrowser` instance would it be better if this class implemented `IDisposable` and was disposed with the `ChromiumWebBrowser` instance? (Should be able to override `Dispose` on the `OffscreenBrowser` class